centreon-engine/src/host.cc

```diff
@@ -3044,12 +3039,9 @@ int host::process_check_result_3x(enum host::host_state new_state,
       get_state_type() == hard ? "HARD" : "SOFT", get_current_state(),
       new_state);
 
-  /* get the current time */
-  time(&current_time);
-
   /* default next check time */
-  next_check = (unsigned long)(current_time + (get_check_interval() *
-                                               config->interval_length()));
+  next_check = (unsigned long)(get_last_check() + (get_check_interval() *
+                                                   config->interval_length()));
```

Comment:
Is the cast really needed?
parenthesis are not needed.
Instead of 'unsigned long', uint32_t is better.